### ISSUE TITLE
$300 Bounty: Add Field IDs to n8n Airtable Node

### ISSUE THREAD

**Issue by anglilian** [bluedotimpact/bluedot#19]
I’d like to request an enhancement for the [Airtable node](https://docs.n8n.io/integrations/builtin/app-nodes/n8n-nodes-base.airtable/) in n8n. Currently, the node requires field names when creating or updating records, but Airtable’s API uses field IDs. This can cause issues, especially when field names change or contain special characters, breaking workflows.

A similar functionality for targeting tables by ID instead of table name has been discussed and implemented (see [this thread](https://community.n8n.io/t/airtable-node-target-table-with-table-name-instead-of-id/26067)), and having a similar feature for fields would be extremely useful. There was a similar feature request [here 1](https://community.n8n.io/t/airtable-node-option-to-allow-for-using-field-ids/29298), but it hasn’t received much traction.

**Proposed Solution:**
- Add an option for the Airtable node to accept field IDs instead of field names for creating and updating records.
- In the n8n UI, the fields should still display the field names for easy configuration, but the underlying logic should use the field IDs, making workflows more resilient to Airtable schema changes.

**Functional Requirements**
- The Airtable node should allow field IDs for both the “Create” and “Update” operations.
- The UI should display field names during configuration but use field IDs in API calls.

**Compatibility & Documentation**
- Ensure the feature supports all Airtable data types and includes proper error handling.
- Update documentation to explain how to use the feature and any toggle options for field name vs. field ID.

**Testing**
- Provide test cases showing the feature works with field name changes and special characters.
- Include a basic workflow test that demonstrates functionality for both single and batch records.

**Review**
- The feature is merged into the main branch and works

Bounty: I am offering $300 for this, but I am open to negotiation based on complexity and feedback. If you are interested or would like clarification, please reach out [anglilian@bluedot.org](mailto:anglilian@bluedot.org)

**Comment by pnogemane** [comment]
Can I pick up the issue?

**Comment by pnogemane** [comment]
close #19

**Comment by domdomegg** [comment]
Hi @pnogemane Yep happy for you to attempt this. We'll pay the bounty once this is merged into the n8n main branch and works :)

**Comment by pnogemane** [comment]
Noted I have reopened the pull request. I am waiting for the reviews to be completed and I will advice as soon as the branch is tested and merged

**Comment by pnogemane** [comment]
> Hi @pnogemane Yep happy for you to attempt this. We'll pay the bounty once this is merged into the n8n main branch and works :)

![Screenshot 2024-11-12 030858](https://github.com/user-attachments/assets/66a9b9d8-4496-45a5-80dc-cce045fb94b0)

The branch has been scheduled for review. I will keep you updated on how it goes :)

**Comment by pnogemane** [comment]
Hi @anglilian anglilian I have implemented the required changes and the branch has been scheduled for automatic merge. 

![Able to Merge](https://github.com/user-attachments/assets/2290f69c-9620-4e93-984c-3424f0c6ad75)

**Comment by domdomegg** [comment]
Great thanks for the update that it is ready to merge! We'll pay the bounty once it is _actually_ merged, as it will require approval by the n8n team etc. :)

**Comment by pnogemane** [comment]
See pull Request : https://github.com/pnogemane/n8n-Add-option-to-use-field-IDs-in-Airtable-node/pull/2

**Comment by domdomegg** [comment]
Hi @pnogemane, we'll pay the bounty once this is merged into the n8n main branch and works :)

I believe you've merged it into your own copy of the repo, not the n8n branch. Merging it into n8n will likely require the approval of the n8n maintainers.

**Comment by pnogemane** [comment]
Hello @domdomegg  

I hope you are well. This is the final stage of the work. I sent through a code review and it was reviewed and approved by the maintainers.

The changes had to then wait in line before being merged to the n8n branch. But finally they where. At this point I think its fair to pay. Thank you and much appriciated.

**Comment by domdomegg** [comment]
Hi @pnogemane, as far as I can see [you appear to have merged a branch into your own repo](https://github.com/pnogemane/n8n-Add-option-to-use-field-IDs-in-Airtable-node/pull/2), but [the one into the n8n branch has not been merged](https://github.com/n8n-io/n8n/pull/11666). I might have misunderstood what has happened - could you clarify where the code has been merged into the n8n branch itself?

![image](https://github.com/user-attachments/assets/d87aac51-3934-4ee4-b000-29af59b7dcb4)

**Comment by aibozo** [comment]
opened a PR to address this. ill let you know when it gets reviewed.

**Comment by aibozo** [comment]
you should really close this while they aren't taking more prs so people don't waste time on something they are currently reviewing.
